pelicun/model/damage_model.py

```diff
@@ -1296,6 +1304,74 @@ def map_ds(values: np.ndarray, offset: int) -> np.ndarray:
 
             return ds_id
 
+        def parse_scaling_specification(scaling_specification: dict) -> dict:  # noqa: C901
+            """
+            Parse and validate the scaling specification.
+
+            Parameters
+            ----------
+            scaling_specification : dict
+                The scaling specification to be parsed and validated.
+
+            Returns
+            -------
+            dict
+                The parsed and validated scaling specification.
+
+            Raises
+            ------
+            ValueError
+                If the scaling specification is invalid.
+            TypeError
+                If the type of an entry is invalid.
+            """
+            # if there are contents, ensure they are valid.
+            # See docstring for an example of what is expected.
+            parsed_scaling_specification = {}
+            # validate contents
+            for key, value in scaling_specification.items():
+                # loop through limit states
+                if key not in parsed_scaling_specification:
+                    parsed_scaling_specification[key] = {}
+                if 'ALL' in value:
+                    if len(value) > 1:
+                        msg = (
+                            f'Invalid entry in scaling_specification: '
+                            f'{value}. It should only have one entry for `ALL`.'
+                        )
+                        raise ValueError(msg)
+                for LS, specifics in value.items():  # noqa: N806
+                    css = 'capacity adjustment specification'
+                    if not isinstance(specifics, list):
+                        specifics_list = [specifics]
+                    else:
+                        specifics_list = specifics
+                    for spec in specifics_list:
+                        if not isinstance(spec, str):
+                            msg = (
+                                f'Invalud entry in {css}: {spec}. It has to be a string. '
+                                f'See docstring of DamageModel._create_dmg_RVs.'
+                            )
+                            raise TypeError(msg)
+                        capacity_adjustment_operation = spec[0]
+                        number = spec[1::]
+                        if capacity_adjustment_operation not in {'+', '-', '*', '/'}:
+                            msg = f'Invalid operation in {css}: '
+                            raise ValueError(
+                                msg
+                                # f'{capacity_adjustment_operation}'
```

Comment:
I think it was a good idea to show the operation in the message. It might need an escape character before to make sure the actual symbol is shown in the output.

Reply:
Modified